### DISCUSSION TITLE
How does ZIP Archive scan works in winget?

### DISCUSSION THREAD

**Discussion by kimwonj77** [microsoft/winget-cli#3056] [Q&A] [answered]
I was trying to submit pypy package to my package. But local winget and Azure Pipeline winget says 'malware detected'. See microsoft/winget-pkgs#97132

However, In my local machine, Microsoft Defender on Windows 11 (not from winget. By context menu at zip file) nor other third-party malware scan tools like [VirusTotal(result)](https://www.virustotal.com/gui/url/d9a6ea89a02b8521d379b796858c9120e4c52627dff067ecf133dbdf63c619d8) says there's no malicious code in zip file.

I suspect that this was happend due to excutables are not signed by PyPy Team. But it's weird that other zip files (which includes submitted to community repo by me) without signed excutable runs fine.

For who want, download link was: https://downloads.python.org/pypy/pypy3.9-v7.3.11-win64.zip
Which was from: https://www.pypy.org/download.html

**Comment by Trenly** [comment, marked as answer]
There is a malware scan performed when installing a local manifest, which uses the Pure library. This doesn't occur when installing applications from a source like winget because they have already been validated by the pipelines. 

In this case, the validation pipelines always install manifests as local manifests, and so the malware scan is performed inaccurately. The team is currently working to disable the local-archive-malware-scan for runs in the pipelines to allow these programs to be validated using the validation suite instead of the built-in local validation which isn't as accurate
